website/docs/advanced-concepts/version-control-with-git/environments-with-git.md

```diff
@@ -1,41 +1,60 @@
 
-<dd>
+:::note
+If you're using two different self-hosted instances for production and staging, take a backup of both instances before updating Appsmith. Update only the staging instance and test all your apps on it before updating the production instance. In the event of an error, you can rollback by following the instructions on [Restore Appsmith Instance](/getting-started/setup/instance-management/appsmithctl#restore-instance)
+:::
 
-:::info 
-If you are an enterprise user, you can set a [Default branch](/advanced-concepts/version-control-with-git/working-with-branches#default-branch) and [Branch protection](/advanced-concepts/version-control-with-git/working-with-branches#branch-protection), ensuring code quality, collaboration, and stability in a collaborative development environment. 
+## Setup instances
 
-:::
+Follow these steps to set up multiple environments for your app:
 
-</dd>
+1. Create separate workspaces for each phase, such as staging and production.
+
+2. Create an app within the `staging` workspace and [connect to a Git repository](/advanced-concepts/version-control-with-git/connecting-to-git-repository). Once connected, check out a `staging` branch from the default branch.
+
+3. Navigate to the Appsmith homepage and [Import the app](/advanced-concepts/version-control-with-git/import-from-repository) into your `production` workspace using the **Import from Git** feature.
 
-3. Import the app into your production workspace using [Import from Git](/advanced-concepts/version-control-with-git/import-from-repository).
 4. Once the import is complete, add your production datasource configurations in the **Reconnect Datasources** Modal.
 
-### Build and deploy app
-1. Build the app in the staging workspace with staging datasources. 
-2. Use the Git workflow to make changes in the staging branch(or feature branches), before merging to the production branch through a pull request.
-3. When you're ready to deploy, [pull the changes](/advanced-concepts/version-control-with-git/merging-branches#pull-changes) on the production branch in the production workspace app.
 
-### Share app
-1. Invite end users to your production app with the *App Viewer* role. They only have access to view the deployed app.
-2. Invite a few trusted individuals with *Administrator* roles to the production app to setup datasources and deploy the app periodically.
-3. Invite your developers to the staging app with the *Developer* role to build the app and test out changes on the staging datasources.
-4. QA team members can be invited to both staging and production apps with the *App Viewer* role to test out the changes deployed.
+5. Protect the `master` and `staging` branches, with `staging` set as the [Default branch](/advanced-concepts/version-control-with-git/working-with-branches#default-branch), and [protect](/advanced-concepts/version-control-with-git/working-with-branches#branch-protection) the `master` branch in the production workspace.
```

Comment:
Sorry, this is wrong. Add this as part of connect and import flows.

1. Protect Staging and Master branches in staging instances and make staging branch default.
2. Protect Master branch in protection and make `master` branch default.